src/lib/optimizer/join_graph_edge.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <iostream>
+#include <memory>
+
+#include "boost/dynamic_bitset.hpp"
+#include "boost/functional/hash.hpp"
+
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+class AbstractExpression;
+
+/**
+ * A bitset that represents a subsets of the vertices of a JoinGraph.
+ *
+ * We use a dynamic bitset here, since it has good support for operations commonly performed on join vertex sets, such
+ * as union, intersection and subtraction.
+ */
+using JoinGraphVertexSet = boost::dynamic_bitset<>;
+
+/**
+ * Represents a (hyper)edge in a JoinGraph.
+ *
+ * Each predicate must operate exactly on the vertices in vertex_set. That is, each predicate must reference columns
+ * from all vertices in vertex_set and no columns from vertices not in vertex_set. If the predicate wouldn't, then it
+ * would belong to another edge.
```

Comment:
Enforce in constructor?

Reply:
Cannot be done, the JoinGraphEdge has no idea which LQPNode a vertex index refers to, and rightly so.

Reply:
Got it. Maybe add this as a comment?

Reply:
Done.